frontend/ts-src/signup.ts

```diff
@@ -0,0 +1,614 @@
+            {
+                name: "Theme 4",
+                imgUrl: "https://images.unsplash.com/photo-1682685796002-e05458d61f07?ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=3271&q=80",
+            },
+        ],
+    },
+    {
+        key: "bio",
+        text: "Add your bio (you can change this later)",
+        answer: "",
+    },
+];
+
+
+export function init(): RegisterConfig {
+    /* Initialize global object.
+    *
+    * @returns { RegisterConfig }
+    */
+
+    const usernameDetails: string = "username";
+    const username: SetupQuestion | undefined = questions.find((q: SetupQuestion) => q.key === usernameDetails);
+    const userInput = document.getElementById("userInput") as HTMLInputElement;
+    const displayUserInput = document.getElementById("chat-container") as HTMLDivElement;
+
+    return {
+        questionIndex: 0,
+        typingSpeed: 100,
+        username: username,
+        userInput: userInput,
+        displayUserInput: displayUserInput,
+        userTyping: false,
+        /*
+        Dont ask user the next question as they have entered a
+        response which is invalid. If blocked === true then
+        the next question does not get asked.
+        */
+        blocked: false,
+        /* This will hold the data we are going to post to the backend. */
+        userResponses: {},
+    };
+
+}
+
+
+export function nextQuestion(): void {
+    /* Ask user next question.
+    *
+    * @returns { void }
+    */
+    if (registerConfig.questionIndex < questions.length) {
+        const currentQuestion: SetupQuestion = questions[registerConfig.questionIndex];
+        displayMessage("system", currentQuestion.text);
+    }
+}
+
+
+export async function processUserInput(): Promise<void> {
+    /* Process incoming user input.
+    *
+    * @returns { void }
+    */
+
+    if (registerConfig.questionIndex >= questions.length) { return; }
+
+    const currentQuestion: SetupQuestion = questions[registerConfig.questionIndex];
+    const key: string = currentQuestion.key;
+    const userReply: string = registerConfig.userInput.value;
+
+    switch(key) {
+        case QuestionKeys.EMAIL: {
+            registerConfig.blocked = !isValidEmail(userReply);
+            break;
+        }
+        case QuestionKeys.USERNAME: {
+            const usernameAvailable = await usernameIsAvailable(clean(userReply));
+            registerConfig.blocked = !usernameAvailable ? true : !isValidUsername(userReply);
+            break;
+        }
+        case QuestionKeys.PASSWORD: {
+            registerConfig.blocked = !isValidPassword(userReply);
+            break;
+        }
+        case QuestionKeys.DOMAINNAME: {
+            const domainNameAvailable: boolean = await domainIsAvailable(clean(userReply));
+            registerConfig.blocked = !domainNameAvailable ? true : !isValidDomainName(userReply);
+            break;
+        }
+        case QuestionKeys.THEME: {
```

Comment:
You can omit the braces and chain the case statements.
```
case QuestionKeys.THEME:
case QuestionKeys.TITLE:
case QuestionKeys.BIO:
default:
  break
```

Reply:
Especially as below its used correctly

https://github.com/MooseyAnon/corna/pull/10/commits/d3381307338db0af4217dbb1f14418973701b159#diff-843480ef208180a931162cb56636b06943ad03de6b080d6d10b9a6454ba81c37R283-R292

Reply:
Its for "safe" block scoping. ESLint complains if you set a variable in more than one case block without curley braces.

Also curly braces in switch statements are relatively common (although ugly).